remote-config/src/main/kotlin/com/rakuten/tech/mobile/remoteconfig/verification/ConfigVerifier.kt

```diff
@@ -0,0 +1,44 @@
+package com.rakuten.tech.mobile.remoteconfig.verification
+
+import android.util.Log
+import androidx.annotation.VisibleForTesting
+import com.rakuten.tech.mobile.remoteconfig.Config
+import com.rakuten.tech.mobile.remoteconfig.toInputStream
+
+internal class ConfigVerifier @VisibleForTesting constructor(
+    private val cache: PublicKeyCache,
+    private val signatureVerifier: SignatureVerifier
+) {
+
+    constructor(cache: PublicKeyCache) : this(cache, SignatureVerifier())
+
+    fun verify(config: Config) : Boolean {
+        return try {
+            val publicKey = cache[config.keyId] ?: return false
+
+            val isVerified = signatureVerifier.verify(
+                publicKey,
+                config.values.toInputStream(),
+                config.signature
+            )
+
+            if (isVerified) {
+                true
```

Comment:
The return is implicit from the return before the try/catch?

Reply:
Yep, try/catch can be used as an expression.